nexus/peer-mysql/src/lib.rs

```diff
@@ -0,0 +1,225 @@
+mod ast;
+mod client;
+mod stream;
+
+use std::fmt::Write;
+
+use peer_cursor::{
+    CursorManager, CursorModification, QueryExecutor, QueryOutput, RecordStream, Schema,
+};
+use pgwire::error::{ErrorInfo, PgWireError, PgWireResult};
+use pt::peerdb_peers::MySqlConfig;
+use sqlparser::ast::{CloseCursor, Declare, FetchDirection, Statement};
+use stream::MyRecordStream;
+
+pub struct MySqlQueryExecutor {
+    peer_name: String,
+    client: client::MyClient,
+    cursor_manager: CursorManager,
+}
+
+impl MySqlQueryExecutor {
+    pub async fn new(peer_name: String, config: &MySqlConfig) -> anyhow::Result<Self> {
+        let mut opts = mysql_async::OptsBuilder::default().prefer_socket(Some(false)); // prefer_socket breaks connecting to StarRocks
+        if !config.user.is_empty() {
```

Comment:
if it is empty should the client even be constructed, similar for other fields

Reply:
Yes, it'll use defaults (in this case, `root`)

Reply:
should we make username etc. optional in that case in the analyzer logic?

Reply:
Yes